OpenDreamRuntime/Procs/DMOpcodeHandlers.cs

```diff
@@ -1887,7 +1887,18 @@ private static void HandleSuffixPronoun(ref StringBuilder formattedString, ReadO
             }
 
             if (value.TryGetValueAsString(out var refString)) {
-                state.Push(state.DreamManager.LocateRef(refString));
+                var refValue = state.DreamManager.LocateRef(refString);
+                if(container is not DreamObjectWorld && containerList is not null) { //if it's a valid ref, it's in world, we don't need to check
+                    foreach (var containerItem in containerList.GetValues()) {
+                        if (IsEqual(containerItem, refValue)) {
+                            state.Push(refValue);
+                            return ProcStatus.Continue;
+                        }
+                    }
+                    state.Push(DreamValue.Null);
```

Comment:
You can use `ContainsValue()` for this
```suggestion
                    state.Push(containerList.ContainsValue(refValue) ? refValue : DreamValue.Null);
```